src/app/app.component.ts

```diff
@@ -96,38 +133,32 @@ export class AppComponent {
   /**
    * Trials on the current page.
    */
-  public selectedPageTrials: Partial<Trial>[];
+  public selectedPageTrials: ResearchStudySearchEntry[];
   /**
    * The number of items per page.
    */
   private itemsPerPage = 10;
-  /**
-   * Loaded conditions from the patient.
-   */
-  public conditions: Condition[];
-  /**
-   * Conditions loaded from TrialScope.
-   */
-  public trialScopeConditions: string[];
   /**
    * The Angular form model for search options.
    */
-  searchReqObject: { zipCode: string | null; travelRadius: string | null; phase: string; recruitmentStatus: string } = {
+  searchReqObject: SearchFields = {
     zipCode: null,
     travelRadius: null,
-    phase: 'any',
-    recruitmentStatus: 'all',
+    phase: null,
+    recruitmentStatus: null,
   };
-  /*
-     variable for gathering patient bundle resources
-  * */
-  public bundleResources: any = [];
-
-  constructor(private spinner: NgxSpinnerService, private trialScopeService: TrialScopeService, private fhirService: ClientService, private convertService: ConvertCodesService) {
-    this.phaseDropDown = ["Early Phase 1", "Phase 1", "Phase 2", "Phase 3", "Phase 4"];
-    this.recDropDown = ["ACTIVE_NOT_RECRUITING", "COMPLETED", "ENROLLING_BY_INVITATION", "NOT_YET_RECRUITING", "RECRUITING", "SUSPENDED", "TERMINATED", "UNKNOWN", "WITHDRAWN"];
-
+  /**
+   * Patient bundle resources from the FHIR client.
+   */
+  public bundleResources: fhirclient.FHIR.BundleEntry[] = [];
 
+  constructor(private spinner: NgxSpinnerService, private searchService: SearchService, private fhirService: ClientService) {
+    this.phaseDropDown = Object.values(ResearchStudyPhase).map((value) => {
+      return new DropDownValue(value, ResearchStudyPhaseDisplay[value]);
+    });
+    this.recDropDown = Object.values(ResearchStudyStatus).map((value) => {
+      return new DropDownValue(value, ResearchStudyStatusDisplay[value]);
```

Comment:
Not totally sure what is happening but it doesn't look like it is listing the display values from the enum. It is listing the ResearchStudyStatus enum values. Meaning, when I click on the Recruitment filter, I see 'active' instead of 'Active'

Reply:
Works for me? I see the display values in the drop down, and they display when I choose them. What it's doing is pulling all the values (that is, the FHIR values) out of the enum, and then using those to lookup the display values in the ResearchStudyStatusDisplay object. (Which possibly should be named in all caps because it's really a constant, but I dunno - that looks ugly.)

The enum itself doesn't have display values, it simply maps constants to strings, as well as telling TypeScript that for any variable of that type, that set of strings are the only valid values. It's done that way so you can use `ResearchStudyType.ACTIVE` to indicate the `active` study type.

Reply:
I see, it was a misunderstanding on my part what stage this data mapping was being used.